client/cli/src/params/psk_key_params.rs

```diff
@@ -0,0 +1,93 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use clap::Args;
+use sc_network::config::PreShareKeyConfig;
+use std::path::PathBuf;
+
+use crate::{error};
+
+const PRE_SHARED_KEY_FILE: &str = "pre_shared_key";
+
+/// Parameters used to create the `PreSharedKeyConfig`, which determines the pre shared key
+/// used for libp2p networking.
+#[derive(Debug, Clone, Args)]
+pub struct PreSharedKeyParams {
+	/// Path to file with pre-shared key.
+	#[clap(long, value_name = "FILE")]
+	pub psk_key_file: Option<PathBuf>,
+}
+
+impl PreSharedKeyParams {
+	/// Create a `PreSharedKeyConfig` from the given `PreSharedKeyParams` in the context
+	/// of an optional network config storage directory.
+	pub fn pre_shared_key(&self, net_config_dir: &PathBuf) -> error::Result<PreShareKeyConfig> {
+		Ok(
```

Comment:
Do we need to wrap it in result? Is error possible?

Reply:
ok